Update Release Notes for release 13.0.11 (#1682)
chef/chef-server: RELEASE_NOTES.md

## 13.0.11 (2019-06-19)

### Chef Server is now Chef Infra Server

Chef Server has a new name, but don’t worry, it’s the same Chef Server you’ve grown used to. You’ll notice new branding throughout the application and documentation but the command `chef-server-ctl` remains the same.

### Chef EULA

Chef Infra Server requires an EULA to be accepted by users before it can be installed. Users can accept the EULA in a variety of ways:

- `chef-server-ctl reconfigure --chef-license accept`
- `chef-server-ctl reconfigure --chef-license accept-no-persist`
- `CHEF_LICENSE="accept" chef-server-ctl reconfigure`
- `CHEF_LICENSE="accept-no-persist" chef-server-ctl reconfigure`

Finally, if users run `chef-server-ctl reconfigure` without any of these options, they will receive an interactive prompt asking for license acceptance. If the license is accepted, a marker file will be written to the filesystem unless `accept-no-persist` is specified. Once this marker file is persisted, users no longer need to set any of these flags.

See our [Frequently Asked Questions document](https://www.chef.io/bmc-faq/) for more information on the EULA and license acceptance.

### Deprecation notice

> - Deprecated PowerPC and s390x platforms. (https://blog.chef.io/2018/11/01/end-of-life-announcement-for-chef-server-for-linux-on-ibm-z-and-linux-on-ibm-power-systems/)

Reviewer: ```suggestion
- [Deprecated PowerPC and s390x platforms](https://blog.chef.io/2018/11/01/end-of-life-announcement-for-chef-server-for-linux-on-ibm-z-and-linux-on-ibm-power-systems/)
```

> - Deprecated Keepalived/DRBD based HA. (https://blog.chef.io/2018/10/02/end-of-life-announcement-for-drbd-based-ha-support-in-chef-server/)

Reviewer: ```suggestion
- Deprecated Keepalived/DRBD-based HA. (https://blog.chef.io/2018/10/02/end-of-life-announcement-for-drbd-based-ha-support-in-chef-server/)
```

- Deprecated Ubuntu 14.04 support. (Ubuntu 14 was EoL’d at the end of April 2019)

### Updates and Improvements

> - Updated OpenResty to 1.13.6.2

Reviewer: This needs a blank line between the header (line 36) and the first bullet point (line 37).

Author: Will add it in!

>     - This fixes two CVEs CVE-2018-9230 and CVE-2017-7529.

Reviewer: ```suggestion
    - This fixes two CVEs (CVE-2018-9230 and CVE-2017-7529).
```

    - This version cannot be built on PowerPC and s390x because those platforms are not supported in mainline luajit.
- Updated Ruby version to 2.5.5
- Updated Chef Infra Client to 14.11.21

> - Updated runit to 5.1.1

Reviewer 2: ```suggestion
- Updated runit cookbook to 5.1.1
```

> - Migrate unit tests from Travis to Buildkite. Reorganize them for improved speed, stability and portability.

Reviewer: ```suggestion
- Migrated unit tests from Travis to Buildkite. Reorganized them for improved speed, stability and portability.
```

- Habitat:

>     - Added some Habitat packaging improvements with parameterized search_server.

Reviewer: Is this the only bullet point for the Habitat updates?

Author: yes - looks like it is for that release.

Reviewer: Just wanted to make sure. Thank you!

- Erchef request size increased from 1,000,000 to 2,000,000 bytes to better support inspec scanning via the audit cookbook.

> - Nginx logs cleanup. No longer log 404's in the nginx error log since in the Chef API, 404s are typically not errors. Rather they are often the expected response about an object that doesn't exist. However the logs will continue to show 404's in the request logs.

Reviewer: ```suggestion
- Nginx error logs no longer log 404s. In the Chef API, 404s are typically not errors. Rather, they are often the expected response about an object that doesn't exist. The logs will continue to show 404s in the request logs.
```

> - Rendering profiles and data-collector upstreams correctly if their root_url is configured. If the data_collector token secret is not set, 401 response code and an error message will be seen instead of 404.

Reviewer: ```suggestion
- Profiles and data-collector upstreams now render correctly if their root_url is configured. If the data_collector token secret is not set, a 401 response code and an error message will be seen instead of 404.
```

Reviewer: In my suggestion above, I changed the word order around in this sentence. Is this sentence still accurate in what it describes?

## 12.19.31 (2019-03-07)

> This release contains some Habitat improvements and updates like Openssl in

Reviewer: ```suggestion
This release contains some Habitat improvements and updates, like Openssl in
```

> Habitat and Omnibus release was done to keep the two in sync.

Reviewer: What are the "two in sync" referring to? Additional edits are needed for clarity.

Author: Ahh, the two meant Habitat and Omnibus releases. Will add some clarification. Thanks!

> - chef-server-ctl leverages HAB_LISTEN_CTL envvar if available

Reviewer: ```suggestion
- `chef-server-ctl` leverages HAB_LISTEN_CTL envvar if available.
```

## 12.19.26 (2019-01-31)

> This release contains some minor improvements and updates to included software.

Reviewer: ```suggestion
This release contains some minor improvements and updates to included software:
```

Reviewer: I am not sure if "include" (or "included") is the accurate term to use. I need to think about what my suggested word choice would be.

- Added request id to nginx log to easily track the Chef request through the logs.
- Bundler pinned to 1.17 to avoid taking the 2.0 upgrade.
- Erlang updated to 18.3.4.9
    - Fixed two CVEs CVE-2017-1000385 and CVE-2016-10253. SSL headers got stricter which unfortunately broke LDAP. (Issue #1642)

>     - Remove et, debugger, gs and observer as they depend on wx, which is not available on all platforms.

Reviewer: ```suggestion
    - Removed et, debugger, gs and observer as they depend on wx, which is not available on all platforms.
```

- Ruby updated to 2.5.3.
- Chef Client updated to 14.5.
- Erchef and Bookshelf can optionally use mTLS protocol for their internal communications.

> - Add configuration for pedant SSL signed requests to include mTLS support.

Reviewer: ```suggestion
- Added configuration for pedant SSL-signed requests to include mTLS support.
```

- Habitat package improvements:

>     - Increased authn:keygen_timeout for oc_erchef hab pkg.

Reviewer: ```suggestion
    - Increased `authn:keygen_timeout` amount for `oc_erchef` hab pkg.
```

>     - Removed do_end function from chef-server-ctl hab plan.

Reviewer: ```suggestion
    - Removed `do_end` function from `chef-server-ctl` hab plan.
```

>     - Enhanced chef-server-ctl to function in more habitat environments.

Reviewer: ```suggestion
    - Enhanced `chef-server-ctl` to function in more habitat environments.
```

>     - chef-server-ctl commands pass relevant TLS options during bifrost API calls.

Reviewer: ```suggestion
    - `chef-server-ctl` commands pass relevant TLS options during bifrost API calls.
```

> - Used standard ruby-cleanup definition that shrinks install size by ~5% on disk.

Reviewer: ```suggestion
- Used standard ruby-cleanup definition, which shrinks install size by ~5% on disk.
```
